### PR TITLE
improve and deduplicate types

### PR DESCRIPTION
**Pull request by dwelle**
As discussed, I DRYed out the docs by moving them to the type file, and referenced them in JSDoc for ease of use. I've also fixed some types that were supposed to be optional.

I'd say let's migrate to TS, but I have no experience in using TS with ESM packages.

### CODE REVIEW

File: src/nativefs/file-save-nativefs.mjs [hunk 1]
```diff
@@ -16,15 +16,10 @@
...
- * @param {Object} [options] - Optional options object.
- * @param {string} options.fileName - Suggested file name.
- * @param {string} options.description - Suggested file description.
- * @param {string[]} options.extensions - Suggested file extensions.
- * @param {FileSystemHandle} [handle] - Optional file handle to save in place.
+ * Saves file to disk using the (legacy) Native File System API.
+ * @type { typeof import("../../index").fileSave }
  */
-export default async (blob, options = {}, handle = null) => {
+export default async function fileSave (blob, options = {}, handle = null) {
```
Comment (tomayac) [inline src/nativefs/file-save-nativefs.mjs:22]: Can we not do this here, or do it everywhere else, too? I have no strong opinion either way, just would like consistency.

Comment (dwelle) [reply src/nativefs/file-save-nativefs.mjs:22]: oops, that's a leftover from trying to make TS work :D. Will remove.

**Comment by tomayac** [comment]
As you are de-DRY-ing, I also noticed the file names are DRY. The folder has the info we need, no need to repeat it in the file names again:

`/src/legacy/file-open-legacy.mjs` → `/src/legacy/file-open.mjs`

Do you want to take care of this in the PR, too?

**Comment by tomayac** [comment]
Re: https://github.com/GoogleChromeLabs/browser-nativefs/pull/15/commits/1dc49c4ff10f72ccb3cddc6c13daa078f5f31910: The same applies for "nativefs" in the file names. Can you rename those, too?

**Comment by dwelle** [comment]
Missed. Done.

**Comment by tomayac** [comment]
Thanks :-) Getting there! While reviewing, I realized the `isFile` method is no longer there, it's now a [`kind`](https://github.com/WICG/native-file-system/blob/master/changes.md#user-content-determining-if-a-handle-is-a-file-or-directory:~:text=handle.kind) attribute. Could you do a sanity check of [the rest of the types](https://github.com/GoogleChromeLabs/browser-nativefs/blob/5246718f6ef19abf868142b61e56093840c4193b/index.d.ts#L49-L70), too?

**Comment by dwelle** [comment]
Will do later today.

**Comment by tomayac** [comment]
> Will do later today.

Thanks a ton! 👏

**Comment by dwelle** [comment]
>  I realized the `isFile` method is no longer there

It's still there for the OT, right? Either way, it's not shipped yet so do we want to update the interface yet? Anyway, I'll finish this tomorrow as I got held up in other work.

**Comment by tomayac** [comment]
> > I realized the `isFile` method is no longer there
> 
> It's still there for the OT, right? Either way, it's not shipped yet so do we want to update the interface yet? Anyway, I'll finish this tomorrow as I got held up in other work.

It's live as of Chrome 86. Not sure if there is a way to express that it can expose either the one, or the other API shape.

**Comment by dwelle** [comment]
I went with using only the latest spec.

Unrelated: why does the library export a demo function `imageToBlob`?

**Comment by tomayac** [comment]
> I went with using only the latest spec.

Thanks, this looks good.

> Unrelated: why does the library export a demo function `imageToBlob`?

It's needed in the demo and may come in handy as a general convenience function more people would use.

**Comment by dwelle** [comment]
> may come in handy as a general convenience function more people would use.

That doesn't sound like a too strong an argument, it's not even in the readme, and I'd not bloat the library by unrelated util helpers. Just my 2 cents :).

**Comment by tomayac** [comment]
> > may come in handy as a general convenience function more people would use.
> 
> That doesn't sound like too strong an argument, it's not even in the readme, and I'd not bloat the library by unrelated util helpers. Just my 2 cents :).

That's fair. I'll move it to the demo folder and remove if from the lib in a separate PR.

**Pull request merged**